src/metabase/models/setting.clj

```diff
@@ -242,7 +242,13 @@
 (defn get-json
   "Get the string value of `setting-definition-or-name` and parse it as JSON."
   [setting-definition-or-name]
-  (json/parse-string (get-string setting-definition-or-name) keyword))
+  (try
+    (json/parse-string (get-string setting-definition-or-name) keyword)
+    (catch Throwable e
+      (let [{setting-name :name} (resolve-setting setting-definition-or-name)]
+        (throw (ex-info (tru "Error parsing JSON setting {0}: {1}" setting-name (ex-message e))
+                        {:setting setting-name}
+                        e))))))
```

Comment:
Added context to make failures when a Setting was previously encrypted easier to understand